torch/serialization.py

```diff
@@ -330,6 +330,11 @@ def _validate_privateuse1_device(location, backend_name):
                            'please use torch.load with map_location=torch.device(\'cpu\') '
                            'to map your storages to the CPU.')
     device_module = getattr(torch, backend_name)
+    if hasattr(device_module, '_utils'):
+        utils_module = getattr(device_module, '_utils')
+        if hasattr(utils_module, '_get_device_index'):
+            get_device_index_fn = getattr(utils_module, '_get_device_index')
+            device_index = get_device_index_fn(location, True)
```

Comment:
Should the code above setting device_index be in the else statement here?

Reply:
modify else statement， please review